examples/dns/resource_record_set.yaml

```diff
@@ -0,0 +1,16 @@
+apiVersion: dns.gcp.crossplane.io/v1alpha1
+kind: ResourceRecordSet
+metadata:
+  name: example
+  annotations:
+    crossplane.io/external-name: "example.crossplane.io."
+spec:
+  forProvider:
+    kind: dns#resourceRecordSet
+    type: CNAME
+    ttl: 300
+    rrdatas:
+      - "server.example.com."
+    managedZone: crossplane-example-zone
```

Comment:
I believe at some point we will also have `Zone` as managed resource, hence need to introduce [cross-resource referencing support](https://crossplane.io/docs/v1.3/concepts/managed-resources.html#dependencies), but nothing to do yet :)